src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(object) {
+    if (!(object instanceof Carnivore) && object.hidden === false) {
+      object.health -= 50;
+    }
+
+    Animal.alive.forEach((animal, index) => {
+      if (animal.health <= 0) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
Using forEach and splice inside it can lead to incorrect behavior because modifying the array during iteration can cause elements to be skipped. Instead, use the filter method to create a new array without the dead animals.